File: BBSK_base/skyline-master/Punto.cpp
#include "Punto.h"

Punto::Punto(int x, int y){
	this->x = x;
	this->y = y;
	dom = 0;
}

Punto::~Punto(){
	
}

int Punto::getX(){
	return x;
}

int Punto::getY(){
	return y;
}

uint Punto::getDom(){
	return dom;
}

void Punto::setX(int x){
	this->x = x;
}

void Punto::setY(int y){
	this->y = y;
}

void Punto::setDom(uint dom){
	this->dom = dom;
}

int Punto::getDistance(ushort option){
    if(option == MANEHATTAN){
		return getManehattanDistance();
	}
	if(option == EUCLIDEAN){
		return getEuclideanDistance();
	}
	if(option == EQUIS){
		return getXDistance();
	}
	cout << "Invalid distance option " << endl;
	exit(1);
}

void Punto::print(){
    cout << "Punto: (" << x << "," << y << ") Dom: " << dom << " ";
}

int Punto::getManehattanDistance(){
	return (x + y);
}

int Punto::getEuclideanDistance(){
	return (x*x + y*y);
}

int Punto::getXDistance(){
	return x;
}

bool Punto::isDominatedBy(Punto * p){
	return (p->getX() <= this->getX() && p->getY() <= this->getY()) 
        && (p->getX() < this->getX() || p->getY() < this->getY());
}

File: BBSK_base/libs/Punto.java
package libs;

public class Punto{
	private long x;
	private long y;
	public Punto(long x, long y){
		this.x = x;
		this.y = y;
	}
	
	public long getX(){
		return x;
	}
	
	public long getY(){
		return y;
	}
	
	public boolean equals(Object o) {
		if (o == this) {
		  return true;
		}
		if (!(o instanceof Punto)) {
		  return false;
		}
		Punto p = (Punto)o;
		return p.getX() == this.getX() && p.getY() == this.getY();
    } 
	
	
}